test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Binders/MetadataBindingTests.cs

```diff
@@ -0,0 +1,215 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="MetadataBindingTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using FluentAssertions;
+using Microsoft.OData.UriParser;
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+using ODataErrorStrings = Microsoft.OData.Strings;
+
+namespace Microsoft.OData.Tests.UriParser.Binders
+{
+    /// <summary>
+    /// Unit test for the MetadataBinding class.
+    /// </summary>
+    public class MetadataBindingTests
+    {
+        private static readonly ODataUriParserConfiguration configuration = new ODataUriParserConfiguration(HardCodedTestModel.TestModel);
+
+        public MetadataBindingTests()
+        {
+        }
+
+        #region Tests
+        [Fact]
+        public void MetadataBinderProcessSkipWithNullReturnsNull()
+        {
+            long? result = MetadataBinder.ProcessSkip(null);
+            result.Should().Be(null);
+        }
+
+        [Fact]
+        public void MetadataBinderProcessSkipWithNegativeNumberShouldThrow()
+        {
+            Action metadataBinderAction = () => MetadataBinder.ProcessSkip(-1);
+            metadataBinderAction.ShouldThrow<ODataException>().WithMessage(
+                ODataErrorStrings.MetadataBinder_SkipRequiresNonNegativeInteger("-1"));
+        }
+
+        [Fact]
+        public void MetadataBinderProcessSkipWithPositiveNumberShouldReturnNumber()
+        {
+            long? result = MetadataBinder.ProcessSkip(1);
+            result.Should().Be(1);
+        }
+
+        [Fact]
+        public void MetadataBinderProcessTopWithNullReturnsNull()
+        {
+            long? result = MetadataBinder.ProcessTop(null);
+            result.Should().Be(null);

```

Comment:
FluentAssertions appears to not have defined the BeNull() extension for numbers. I've updated other areas to BeNull() though.